### DISCUSSION TITLE
how to create a regular hexagon?

### DISCUSSION THREAD

**Discussion by woosley** [jgraph/drawio#4434] [General]
I am now stuck to create a regular hexagon. 
by default, what is created is like below
 
![image](https://user-images.githubusercontent.com/61531/139384318-59f1d105-2d46-4e60-8b3e-320a8541c3d5.png)
but how can I make it as a "regular hexagon"? I can not find any way t odo this

**Comment by mararad** [comment]
Hi,

just type "hexagon" in Search field. By moving little orange diamond shape you can modify it.

![hexagon](https://user-images.githubusercontent.com/44292306/139420969-685d023e-47e0-4600-868c-df74112a5e3b.JPG)

Regards,

**Comment by woosley** [comment]
the shape can be changed  in this way but it is super hard to make it perfect "regular hexagon"

**Comment by mararad** [comment]
Our engineer created the issue you can track https://github.com/jgraph/drawio/issues/2368

Regards,

**Comment by sehalwas** [comment]
I still don't know how to do it.

**Comment by iskanred** [comment]
\+

**Comment by DrewRay** [comment]
In case someone else comes back to here, you can use the Arrange >Insert> Shape menu to add a custom shape type and for a hexagon you can draw a tilted one with this shape 

```<shape h="300" w="300" aspect="fixed" strokewidth="inherit">
  <background>
  </background>
  <foreground>
    <fillstroke />
    <path>
      <move x="280" y="225" />
      <line x="150" y="300" />
      <line x="20" y="225" />
      <line x="20" y="75" />
      <line x="150" y="0" />
      <line x="280" y="75" />
      <line x="280" y="225" />
      <close />
    </path>
    <stroke />
  </foreground>
</shape>
```

or a flat one with this 

```<shape h="300" w="300" aspect="fixed" strokewidth="inherit">
  <background>
  </background>
  <foreground>
    <fillstroke />
    <path>
      <move x="300" y="150" />
      <line x="225" y="280" />
      <line x="75" y="280" />
      <line x="0" y="150" />
      <line x="75" y="20" />
      <line x="225" y="20" />
      <line x="300" y="150" />
      <close />
    </path>
    <stroke />
  </foreground>
</shape>
```

**Comment by oe3gwu** [comment]
@DrewRay nice but technically your hexagon is not closed speaking strict SVG Script syntax. This works better:

```
<shape h="300" w="300" aspect="fixed" strokewidth="inherit">
  <background> </background>
  <foreground>
    <fillstroke />
    <path>
      <move x="300" y="150" />
      <line x="225" y="280" />
      <line x="75" y="280" />
      <line x="0" y="150" />
      <line x="75" y="20" />
      <line x="225" y="20" />
      <line x="300" y="150" />
      <close /> <!-- This closes the path -->
    </path>
    <stroke />
  </foreground>
</shape>
```

**Reply by DrewRay** [reply]
Thanks, updated mine as well.

**Comment by ierdss** [comment]
If anyone wants to rotate the shape without affecting the text angle. This is here for you.

# Result
<img width="545" height="301" alt="image" src="https://github.com/user-attachments/assets/63341d48-3d4c-4331-9ac6-7db132761a11" />

# Steps
1. Select the hexagon component
<img width="287" height="262" alt="image" src="https://github.com/user-attachments/assets/76f2a0b7-100f-41b2-adff-d2a1cb283ac6" />

2. Select the "Arrange" tab
<img width="241" height="537" alt="image" src="https://github.com/user-attachments/assets/ca4ff052-5861-4043-bb6f-451817a08a89" />

3. Go to Angle
<img width="241" height="71" alt="image" src="https://github.com/user-attachments/assets/b7171a48-401c-4383-b540-6b1fb094535d" />

4. Select "Rotate shape only by 90°"
<img width="209" height="20" alt="image" src="https://github.com/user-attachments/assets/dff4bed9-bbe8-4b85-ba33-b718c6c02e01" />
